File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    res = {}
    answer = 1
    
    for i in clothes:
        if i[1] in res:
            res[i[1]] += 1
        else:
            res[i[1]] = 1

    for k, v in res.items():
        con = v + 1

        answer *= con
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        for(String[] cloth : clothes) {
            String c = cloth[1];
            map.put(c, map.getOrDefault(c, 0) + 1);
        }
        
        int answer = 1;
        
        for(int key : map.values()) {
            answer *= (key+1);
        }
        
        return answer-1;
    }
}